src/CLUENtuplizer.cpp

```diff
@@ -0,0 +1,344 @@
+    "MCParticles", Gaudi::DataHandle::Reader, this};
+  auto mcps = mcp_handle.get();
+  int mcps_primary = 0;
+  float mcp_primary_energy = 0.f;
+  std::for_each((*mcps).begin(), (*mcps).end(),
+                [&mcps_primary, &mcp_primary_energy] (edm4hep::MCParticle mcp) { 
+                  if(mcp.getGeneratorStatus() == 1){
+                    mcps_primary += 1;
+                    mcp_primary_energy = mcp.getEnergy();
+                  }
+                });
+  info() << "MC Particles = " << mcps->size() << " (of which primaries = " << mcps_primary << ")" << endmsg;
+  // If there is more than one primary, skip event
+  if(mcps_primary > 1){
+    warning() << "This event is skipped because there are " << mcps_primary << " primary MC particles." << endmsg;
+    return StatusCode::SUCCESS;
+  }
+
+  DataObject* pStatus  = nullptr;
+  StatusCode  scStatus = eventSvc()->retrieveObject("/Event/CLUECalorimeterHitCollection", pStatus);
+  if (scStatus.isSuccess()) {
+    clue_calo_coll = static_cast<clue::CLUECalorimeterHitCollection*>(pStatus);
+  } else {
+    throw std::runtime_error("CLUE hits collection not available");
+  }
+
+  // Read EB collection for metadata cellID
+  DataHandle<edm4hep::CalorimeterHitCollection> EB_calo_handle {
+    EBCaloCollectionName, Gaudi::DataHandle::Reader, this};
+  EB_calo_coll = EB_calo_handle.get();
+
+  // Read EE collection
+  DataHandle<edm4hep::CalorimeterHitCollection> EE_calo_handle {
+    EECaloCollectionName, Gaudi::DataHandle::Reader, this};
+  EE_calo_coll = EE_calo_handle.get();
+
+  debug() << "ECAL Calorimeter Hits Size = " << int( EB_calo_coll->size()+EE_calo_coll->size() ) << endmsg;
+
+  // Read cluster collection
+  DataHandle<edm4hep::ClusterCollection> cluster_handle {  
+    ClusterCollectionName, Gaudi::DataHandle::Reader, this};
+  cluster_coll = cluster_handle.get();
+
+  // Get collection metadata cellID which is valid for both EB, EE and Clusters
+  auto collID = EB_calo_coll->getID();
+  const auto cellIDstr = EB_calo_handle.getCollMetadataCellID(collID);
+  const BitFieldCoder bf(cellIDstr);
+  cleanTrees();
+
+  std::uint64_t ch_layer = 0;
+  std::uint64_t nClusters = 0;
+  float totEnergy = 0;
+  float totEnergyHits = 0;
+  std::uint64_t totSize = 0;
+  bool foundInECAL = false;
+
+  for (const auto& cl : *cluster_coll) {
+    m_clusters_event->push_back (evNum);
+    m_clusters_energy->push_back (cl.getEnergy());
+    m_clusters_size->push_back (cl.hits_size());
+
+    m_clusters_x->push_back (cl.getPosition().x);
+    m_clusters_y->push_back (cl.getPosition().y);
+    m_clusters_z->push_back (cl.getPosition().z);
+
+    // Sum up energy of cluster hits and save info
+    // Printout the hits that are in Ecal but not included in the clusters
+    int maxLayer = 0;
+    for (const auto& hit : cl.getHits()) {
+      foundInECAL = false;
+      for (const auto& clEB : *EB_calo_coll) {
+          if( clEB.getCellID() == hit.getCellID()){
+            foundInECAL = true;
```

Comment:
```suggestion
            foundInECAL = true;
            break;  // Found in EB, break the loop
```
No need for the next if